#! /bin/sh -e
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Change in session id broke new window function.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch)   ( cd BUILD/sql-ledger && patch $patch_opts -p3 ) < $0 ;;
       -unpatch) ( cd BUILD/sql-ledger && patch $patch_opts -p3 -R ) < $0 ;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@
diff -Nru /tmp/P47HhTUMbJ/sql-ledger-2.6.18/menu.ini /tmp/iMvQ6ZGqR0/sql-ledger-2.6.19/menu.ini
--- /tmp/P47HhTUMbJ/sql-ledger-2.6.18/menu.ini	2006-05-25 21:29:10.000000000 +0200
+++ /tmp/iMvQ6ZGqR0/sql-ledger-2.6.19/menu.ini	2006-09-11 23:05:07.000000000 +0200
@@ -900,7 +900,6 @@
 module=menu.pl
 action=display
 main=company_logo
-duplicate=1
 target=_blank
 
 [Logout]
diff -Nru /tmp/P47HhTUMbJ/sql-ledger-2.6.18/am.pl /tmp/iMvQ6ZGqR0/sql-ledger-2.6.19/am.pl
--- /tmp/P47HhTUMbJ/sql-ledger-2.6.18/am.pl	2006-09-09 17:10:06.000000000 +0200
+++ /tmp/iMvQ6ZGqR0/sql-ledger-2.6.19/am.pl	2006-09-11 22:52:07.000000000 +0200
@@ -166,42 +166,42 @@
 	  $cookie{$name} = $value;
 	}
 	
-	if ($form->{action} ne 'display') {
-	  if ($cookie{"SL-$form->{login}"}) {
-	    
-	    $form->{sessioncookie} = $cookie{"SL-$form->{login}"};
-	    
-	    $s = "";
-	    %ndx = ();
-            # take cookie apart
-	    $l = length $form->{sessioncookie};
-	    
-	    for $i (0 .. $l - 1) {
-	      $j = substr($myconfig{sessionkey}, $i * 2, 2);
-	      $ndx{$j} = substr($cookie{"SL-$form->{login}"}, $i, 1);
-	    }
-
-	    for (sort keys %ndx) {
-	      $s .= $ndx{$_};
-	    }
-
-            $l = length $form->{login};
-	    $login = substr($s, 0, $l);
-	    $time = substr($s, -10);
-	    $password = substr($s, $l, (length $s) - ($l + 10));
-
-	    # validate cookie
-	    if ((time > $time) || ($login ne $form->{login}) || ($myconfig{password} ne crypt $password, substr($form->{login}, 0, 2))) {
-	      &getpassword(1);
-	      exit;
-	    }
+	if ($cookie{"SL-$form->{login}"}) {
+	  
+	  $form->{sessioncookie} = $cookie{"SL-$form->{login}"};
+	  
+	  $s = "";
+	  %ndx = ();
+	  # take cookie apart
+	  $l = length $form->{sessioncookie};
+	  
+	  for $i (0 .. $l - 1) {
+	    $j = substr($myconfig{sessionkey}, $i * 2, 2);
+	    $ndx{$j} = substr($cookie{"SL-$form->{login}"}, $i, 1);
+	  }
 
-	  } else {
+	  for (sort keys %ndx) {
+	    $s .= $ndx{$_};
+	  }
 
-	    &getpassword(1); 
-	    exit; 
+	  $l = length $form->{login};
+	  $login = substr($s, 0, $l);
+	  $time = substr($s, -10);
+	  $password = substr($s, $l, (length $s) - ($l + 10));
+
+	  # validate cookie
+	  if ((time > $time) || ($login ne $form->{login}) || ($myconfig{password} ne crypt $password, substr($form->{login}, 0, 2))) {
+	    &getpassword(1);
+	    exit;
+	  }
 
+	} else {
+
+	  if ($form->{action} ne 'display') {
+	    &getpassword(1); 
+	    exit;
 	  }
+
 	}
       } else {
 	exit;
